composites/Plugin/Shared/components/Collapsible.js

```diff
@@ -0,0 +1,237 @@
+import PropTypes from "prop-types";
+import React from "react";
+import styled from "styled-components";
+
+import colors from "../../../../style-guide/colors.json";
+import { IconsButton } from "../../Shared/components/Button";
+
+const StyledContainer = styled.div`
+	background-color: ${ colors.$color_white };
+`;
+
+const StyledContent = styled.div`
+	padding: 0 15px;
+`;
+
+const StyledIconsButton = styled( IconsButton )`
+	width: 100%;
+	background-color: ${ colors.$color_white };
+	padding: 15px;
+	justify-content: flex-start;
+	border-color: transparent;
+	border-radius: 0;
+	box-shadow: none;
+	color: ${ colors.$color_button_border_active };
+	
+	svg {
+		&:first-child {
+			margin-right: 8px;
+		}
+		&:last-child {
+			margin-left: 8px;
+		}
+	}
+`;
+
+const StyledTitle = styled.span`
+	white-space: nowrap;
+	text-overflow: ellipsis;
+	overflow-x: hidden;
+	-ms-flex-positive: 1;
+	flex-grow: 1;
+	font-size: 1.03em;
+	font-weight: 600;
+`;
+
+/**
+ * Wraps a component in a heading element with a defined heading level.
+ *
+ * @param {ReactElement} Component    The component to wrap.
+ * @param {int}          headingLevel The heading level.
+ *
+ * @returns {ReactElement} The wrapped component.
```

Comment:
returns a function rather than a react element

Reply:
In this case it should be `ReactComponent`. A higher order component always receives a `ReactComponent` and returns a new `ReactComponent`. `Function` is technically correct, but a function can also be a `ReactComponent`, which it is in this case:

```jsx
const Component = ( props ) => {
    return <div>Hi!</div>;
}
```